sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/AvroGenericRecordToStorageApiProto.java

```diff
@@ -115,34 +120,90 @@ static String convertUUID(Object value) {
   }
 
   static Long convertTimestamp(Object value, boolean micros) {
-    if (value instanceof ReadableInstant) {
-      return ((ReadableInstant) value).getMillis() * (micros ? 1000 : 1);
```

Comment:
This was wrong. BQ always expects epoch microseconds. Conversion should be applied on the raw type depending if it represents millis or micros